Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageCell.swift

```diff
@@ -65,12 +71,45 @@ public class ChatMessageCell: _TableViewCell, ComponentsProvider {
 
         backgroundColor = .clear
         backgroundView = nil
+
+        headerContainerView.backgroundColor = nil
+        footerContainerView.backgroundColor = nil
     }
 
     override public func prepareForReuse() {
         super.prepareForReuse()
 
         messageContentView?.prepareForReuse()
+        headerContainerView.removeFromSuperview()
+        footerContainerView.removeFromSuperview()
+    }
+
+    public func updateDecoration(
+        for decorationType: ChatMessageDecorationType,
+        decorationView: ChatMessageDecorationView?
+    ) {
+        let container = decorationType == .header ? headerContainerView : footerContainerView
+
+        container.subviews.forEach { $0.removeFromSuperview() }
+
+        guard let decorationView = decorationView else {
+            container.removeFromSuperview()
+            return
+        }
+
+        decorationView.translatesAutoresizingMaskIntoConstraints = false
+        container.embed(decorationView)
+        switch decorationType {
+        case .header:
+            containerStackView.insertArrangedSubview(container, at: 0)
+        case .footer:
+            containerStackView.addArrangedSubview(container)
+        }
+
+        container.pin(
```

Comment:
You might end up pinning multiple times if you call updateDecoration on the same cell, right?

Reply:
Yes this is also a good point. @ipavlidakis Any reason why we don't simply hide/show like the previous implementation. Keep in mind that Apple actually recommends this approach for performance reasons, it is better to hide/show views in table view cells, than removing/adding views or changing constraints ad-hoc

Reply:
The issue I had with hide/show is that the spacing between the views is still there, which is causing UI changes. 
What I can do is to remove the spacing from the stackview and move it in the containers instead. WDYT?

Reply:
Yes, that makes sense since if a customer wants to customize the separator, including the spacing, it will only do it in 1 view, it does need to change the spacing in the stack view. So this is more like padding, then spacing actually 👍